File: iOS/DateMate/UI/MyProfile/MyProfileUiState.swift
//
//  MyProfileUiState.swift
//  DateMate
//
//  Created by 홍희표 on 2025/12/01.
//

import Foundation

struct MyProfileUiState {
    var isLoading: Bool = false
    var profile: Profile? = nil
    var stats: UserStats = UserStats(postCount: 0, followers: "", following: "")
    var posts: [Post] = []
    var message: String? = nil
}

struct UserStats {
    var postCount: Int
    var followers: String
    var following: String
}

File: Android/app/src/main/java/com/hhp227/datemate/ui/myprofile/MyProfileUiState.kt
package com.hhp227.datemate.ui.myprofile

import com.hhp227.datemate.data.model.Post
import com.hhp227.datemate.data.model.Profile

data class MyProfileUiState(
    val isLoading: Boolean = false,
    val profile: Profile? = null,
    val stats: UserStats = UserStats(0, "", ""),
    val posts: List<Post> = emptyList(),
    val message: String? = null
)

data class UserStats(val postCount: Int, val followers: String, val following: String)